src/fee.rs

```diff
@@ -0,0 +1,50 @@
+/// This module provides functionality needed to spend a UTXO.
+
+#[cfg(any(test, feature = "rand"))]
+use crate::Utxo;
+
+//One vbyte is equal to four weight units.
+#[cfg(any(test, feature = "rand"))]
+fn weight_to_vbytes(weight: u64) -> u64 {
+    weight / 4
+}
+
+#[cfg(any(test, feature = "rand"))]
+pub fn get_fee(fee_rate: u64, utxo: &impl Utxo) -> u64 {
+    weight_to_vbytes(utxo.get_weight()) * fee_rate
```

Comment:
If you want the library to be robust you might want to do a checked_mul and error if it overflows.

Reply:
I think this is debatable.  The problem with `checked_mul` is that it will return `None` to the calling application instead of a panic which will return a stack trace with potentially useful information to debug.  Another way to look at it is that wouldn't a multiplication that's greater than u64 max be a fatal error?  If the error should not be fatal then I could see wanting to use `checked_mul` and return an `Error` type with a useful message I suppose.

Reply:
I think I'd prefer using [debug_assert](https://doc.rust-lang.org/std/macro.debug_assert.html) to panic possibly with a message.  You're right though that wrapping silently if the result exceeds u64 max is a probably unexpected, and in release mode this could happen, unlike debug mode.

Reply:
My thinking is that as much as possible I expect the library that I use not to panic, but to error if I give them invalid arguments (which would be the case here IMHO with a fee_rate that would be too high). I feel whether the error is fatal or not is for the caller to decide.

Reply:
Ok I see your point.  I suppose it's possible the application allows user input, and doesn't validate the input and then passes extreme values to this lib.  In such a case an error could be returned to prompt the user for a lower fee_rate value (assuming the calling application handles the error).  The other option is just to document that in the library that the application will panic if the values exceed u64 max so the application should validate and send sane values.  Anyway, if you feel that strongly about returning an error that makes some sense and I can add it.

Reply:
I don't feel strongly about it, documenting it would also work but it might be a bit awkward because you can't really give a value since it also depends on the weight.

Reply:
Since `fee_rate` could be user input, I've changed my mind and agree that the application shouldn't crash if a user inputs a crazy value.  It's better if the application validates the input but the lib can also help guard against such occurrences.

Reply:
@Tibo-lg resolved by this commit https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/22/commits/7c6b6343330a600a830c5cf800b4f2a9a875194f

Reply:
the error propagation of https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/22/commits/c9cf487127c2ed8b7f10cc0415cffc15eb25c103 is a bit weird, although I'm not sure of a better way and still using an iterator.  Using an iterator is preferred for performance probably but maybe a standard `for each` would be easier to grok.

Reply:
Yeah I would personally go for code readability first and then consider optimizing later once everything is implemented so you can benchmark to see how big of a difference it makes. Also I wouldn't recommend adding the `num` and `error` dependencies, in general the crates within `rust-bitcoin` try to use as few dependencies as possible.

Reply:
@Tibo-lg ok I've removed the `num` and `error` deps, also I got ride of the iterator `take_while` for now: https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/22/commits/bf7316bec83e0ade6a2d95a7f9992e26ace8e383